### PR TITLE
Change the user and group under which proxysql runs on Debian systems

### PR DESCRIPTION
**Pull request by tullis**
On Debian systems proxysql should be running under the `proxysql` user and group, instead of `root`.
The configuration files should also be owned by this user and group.

I initially started looking to see when the Debian version might have switched from using the `root` account to `proxysql`, so that I could use a version check in `params.pp` but I could not find any time when it did. Therefore I thought it best to merge the case blocks for Ubuntu and Debian into one.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -24,13 +24,7 @@
   $admin_listen_port   = 6032
 
   case $::operatingsystem {
```
Comment (ekohl) [inline manifests/params.pp]: I might make sense to change this to the osfamily instead.

Comment (tullis) [reply manifests/params.pp]: OK. Updating as suggested.

**Comment by tullis** [comment]
Correct. There is currently no `proxysql` package in Debian at all.
[package search results](https://packages.debian.org/search?keywords=proxysql&searchon=names&suite=all&section=all)

File: manifests/params.pp [hunk 2]
```diff
@@ -23,20 +23,14 @@
   $admin_listen_ip     = '127.0.0.1'
   $admin_listen_port   = 6032
 
-  case $::operatingsystem {
+  case $::osfamily {
```
Comment (bastelfreak) [inline manifests/params.pp]: can you please switch this to the new $facts hash? `$facts['os']['family']`

**Comment by mcrauwel** [comment]
The problem is that Percona rebuilds the package and changes the defaults that ProxySQL uses.
ProxySQL has recently released it's own upstream packagemanager repo on http://repo.proxysql.com/

Imho we should add this as the default repo to use, if you want to override the repo to the percona repo you should also override the user/group to run it under...

**Comment by mcrauwel** [comment]
@tullis @ekohl @bastelfreak see #47 

that PR shows how I would update the module currently, I tested that PR on Ubuntu 16.04 and CentOS 7. i still need to review the testsuite and docs but I wanted to push this here already to give an idea...

It's been a while since I wrote puppet-code, my new job doesn't include puppet anymore but I can still help and contribute when time allows...

File: manifests/params.pp (hunk 2 above)
Comment (tullis) [reply manifests/params.pp]: Thanks @bastelfreak - I have updated the code as you requested and squashed the change.

**Comment by tullis** [comment]
Thanks all for your input. I'm happy to close this PR in favour of the default use of the new upstream repo. Does anyone know when the version on the forge might be updated with a new release?

**Comment by tullis** [comment]
Closing in favour of #47

**Pull request closed without merging**